jupyterlab/CORPORATE.md

```diff
@@ -0,0 +1,107 @@
+# Corporate Engagement and Contribution Guide
+
+Along with welcoming contributions from individuals, we also welcome
+contributions to JupyterLab from corporations. Over a number of years of working
+with a wide range of corporations, we have discovered a set of principles and
+practices that enable these collaborations to be healthy, productive, and
+sustainable. These principles and practices are detailed here, along with a
+mental model that sets the stage.
+
+## Mental Model
+
+We would like to offer the following mental model of open-source projects, set
+as a parable. This idea was first introduced by a core Python contributor, Brett
+Cannon, on Twitter:
+
+> Pull requests can be like someone trying to give you a puppy you didn't ask
+> for; they mean well, but they can forget a puppy is a decade-or-more commitment…
+
+Let's expand on this idea a bit more.
+
+An open-source software project is like a big yard full of puppies. Puppies are
+amazing!!! Lots of people stop by to look at the puppies, play with them, take
+them on walks, and play fetch. These people are like the users of an open-source
+project. A few people, the maintainers, live in the yard with the puppies. The
+maintainers feed the puppies, take care of them, wash them, take them to the
+vet, get their shots, and keep the yard clean. Just to keep the existing puppies
+happy and healthy requires a ton of work and money.
+
+Working in a yard full of puppies is CRAZY! At times, there can be a line around
+the block of people wanting to play with the puppies and talk, or complain, to
+the maintainers. Sometimes, people show up wth more puppies. These are like pull
+requests to an open-source project. While the maintainers love puppies and would
+love to be able to have more of them for everyone to enjoy, each new puppy has
+to be cleaned, fed, trained, taken to the vet, etc.
+
+## Practices and Principles
+
+The following principles and practices should be followed by corporations
+wanting to contribute to JupyterLab. These apply to both corporations as a
+whole, as well as individual contributors within those corporations.
+
+**Empower individual contributors.** In the daily development and design work of
+the project, everyone contributes as individuals. Because of this, if your
+corporation wants to engage with and contribute to JupyterLab, the primary means
+will be through allocating, empowering, and encouraging your individual
+employees to contribute in an ongoing manner. Put more bluntly, pay your
+employees to contribute to Jupyter.
+
+**Make balanced contributions.** We expect all contributors to make balanced
+contributions that match their skill level and level of participation. Balanced
+contributions include work on new features, fixing bugs, reviewing pull
+requests, writing documentation, issue triage, answering users' questions,
```

Comment:
> answering users' questions

This is a really good thing to specifically enumerate. Answer questions on channels like Gitter helps expand community.